src/pages/home/uploadSong/AdvancedSongDetails.tsx

```diff
@@ -29,7 +32,10 @@ const AdvancedSongDetails = () => {
     useFormikContext<UploadSongRequest>();
 
   const { data: { date: earliestReleaseDate } = {} } =
-    useGetEarliestReleaseDateQuery();
+    useGetEarliestReleaseDateQuery(undefined, {
+      // endpoint throws error if user hasn't added first name
+      skip: !firstName,
```

Comment:
Should a user even get this far if they didn't enter the required first name from the initial profile creation flow? Same question for the `UploadSong.tsx` change.